locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+      54000       |    6    |     35
+```
+While the two processes start off the same and seem to work the GC at the same
+rate, it becomes evident that after a few seconds with a properly working
+backpressure system in place, it spreads the GC load across consistent
+intervals of 4-8 milliseconds until the end of the data transfer.
+
+However, when a backpressure system is not in place, the V8 garbage collection
+starts to drag out. The normal binary called the GC approximately `75`
+times in a minute, whereas, the modified binary fires only `36` times.
+
+This is the slow and gradual debt accumulating from a growing memory usage. As
+data gets transferred, without a backpressure system in place, more memory is
+being used for each chunk transfer.
+
+The more memory that is being allocated, the more the GC has to take care of in
+one sweep. The bigger the sweep, the more the GC needs to decide what can be
+freed up and scanning for detached pointers in a larger memory space will 
+consume more computing power.
+
+## Memory Exhaustion
+
+To determine the memory consumption of each binary, we've clocked each process
+with `/usr/bin/time -lp sudo ./node ./backpressure-example/zlib.js`
+individually.
+
+This is the output on the normal binary:
+
+```
+  Respecting the return value of .write()
+=============================================
+real        58.88
+user        56.79
+sys          8.79
+  87810048  maximum resident set size
+         0  average shared memory size
+         0  average unshared data size
+         0  average unshared stack size
+     19427  page reclaims
+      3134  page faults
+         0  swaps
+         5  block input operations
+       194  block output operations
+         0  messages sent
+         0  messages received
+         1  signals received
+        12  voluntary context switches
+    666037  involuntary context switches
+```
+
+The maximum byte size occupied by virtual memory turns out to be approximately
+`87.81 mb`.
+
+And now changing the [return value][] of the `.write()` function, we get:
+
+```
+ Without respecting the return value of .write():
+==================================================
+real        54.48
+user        53.15
+sys          7.43
+1524965376  maximum resident set size
+         0  average shared memory size
+         0  average unshared data size
+         0  average unshared stack size
+    373617  page reclaims
+      3139  page faults
+         0  swaps
+        18  block input operations
+       199  block output operations
+         0  messages sent
+         0  messages received
+         1  signals received
+        25  voluntary context switches
+    629566  involuntary context switches
+```
+
+The maximum byte size occupied by virtual memory turns out to be approximately
+`1.52 gb`.
+
+Without streams in place to delegate the backpressure, there is a memory space 
+with an entire degree of magnitude greater that is being allocated. That is a 
```

Comment:
Tbh I went back to edit this many times and couldn't figure out why it sounded so off .. hehe thanks!